### ISSUE TITLE
 HTTP request failed: Protocol "stratum+tcp" not supported or disabled in libcurl

### ISSUE THREAD

**Issue by ZAX6** [ghostlander/nsgminer#34]
hi guys , 
i tried to mine using the nsgminer , had issue on both windows and ubuntu, 
i compiled nsgminer on ubuntu and used the released version on windows and ubuntu aswell , had the issue on all of them.

the output is : 

```
[2018-02-06 03:10:03] Started nsgminer 0.9.4
[03:10:03] DISPLAY not set, setting :0 just in case
[03:10:03] OpenCL platform 0 vendor: Advanced Micro Devices, Inc.
[03:10:03] OpenCL platform 0 name: AMD Accelerated Parallel Processing
[03:10:03] OpenCL platform 0 version: OpenCL 2.0 AMD-APP (2482.3)
[03:10:03] Platform 0 devices: 4
[03:10:03]      0       Hawaii
[03:10:03]      1       Hawaii
[03:10:03]      2       Hawaii
[03:10:03]      3       Hawaii

[03:10:03] OCL 0: Set temperature config: target=89 cutoff=95
[03:10:03] OCL 1: Set temperature config: target=89 cutoff=95
[03:10:03] OCL 2: Set temperature config: target=89 cutoff=95
[03:10:03] OCL 3: Set temperature config: target=89 cutoff=95
[03:10:03] Probing for an alive pool
[03:10:03] Testing pool stratum+tcp://eu1.altminer.net:4233
[03:10:03] Popping work to stage thread
[03:10:03] HTTP request failed: Protocol "stratum+tcp" not supported or disabled in libcurl
[03:10:03] Failed to connect in json_rpc_call
[03:10:03] initiate_stratum with sockbuf=(nil)
[03:10:03] JSON stratum auth failed: (unknown reason)
[03:10:03] Unable to get work from pool 0 stratum+tcp://eu1.altminer.net:4233
[03:10:03] No servers were found that could be used to get work from.
[03:10:03] Please check the details from the list below of the servers you have input
[03:10:03] Most likely you have input the wrong URL, forgotten to add a port, or have not set up workers
[03:10:03] Pool: 0  URL: stratum+tcp://eu1.altminer.net:4233  User: WALLET  Password: c=VIVO
No servers could be used! Exiting.
```

did alot of search on google  , no clue yet

**Comment by fogoat** [comment]
Did you try a different pool?

**Comment by ZAX6** [comment]
yes , didn't work at all , tried different ports as-well, i think the clue to this problem is in this line:
`HTTP request failed: Protocol "stratum+tcp" not supported or disabled in libcurl`

**Comment by ghostlander** [comment]
Stratum+TCP isn't supported by libcurl anyway, but it doesn't matter. Your problem is:

[03:10:03] Failed to connect in json_rpc_call
[03:10:03] initiate_stratum with sockbuf=(nil)
[03:10:03] JSON stratum auth failed: (unknown reason)

**Comment by fogoat** [comment]
@ghostlander can you explain what you mean?

**Comment by ghostlander** [comment]
Authentication failed or the pool returned message the miner couldn't understand.

**Comment by ZAX6** [comment]
do you have any solutions to this problem , or any clue that i could search for it @ghostlander ,

**Comment by ghostlander** [comment]
Since pool ops of these yaamp/yiimp pools are not inclined to investigate and fix the issue on their side, this kind of handshake failure has to be identified and fixed within the miner. I'll add it to my list of things to do, but it may take a while.

**Comment by ghost** [comment]
hi there ,
i am using nsgminer for mining on ubuntu,
i compiled nsgminer on ubuntu and while mining had this issue can anyone help please.

the output is :
[14:04:38] Probing for an alive pool
[14:04:38] Testing pool stratum+tcp://192.168.1.104:3333
[14:04:38] Popping work to stage thread
[14:04:38] HTTP request failed: Protocol stratum+tcp not supported or disabled in libcurl
[14:04:38] Failed to connect in json_rpc_call
[14:04:38] initiate_stratum with sockbuf=(nil)
[14:04:38] Stratum connect failed to pool 0: Failed to connect to 192.168.1.104 port 3333: Connection refused
[14:04:38] Initiate stratum failed
[14:04:38] Unable to get work from pool 0 stratum+tcp://192.168.1.104:3333
[14:04:38] No servers were found that could be used to get work from.
[14:04:38] Please check the details from the list below of the servers you have input
[14:04:38] Most likely you have input the wrong URL, forgotten to add a port, or have not set up workers
[14:04:38] Pool: 0  URL: stratum+tcp://192.168.1.104:3333  User: workeruser  Password: workerpass
[14:04:38] Press any key to exit or the miner will try again in 15s.
[14:04:53] Testing pool stratum+tcp://192.168.1.104:3333
[14:04:53] HTTP request failed: Protocol stratum+tcp not supported or disabled in libcurl
[14:04:53] Failed to connect in json_rpc_call
[14:04:53] initiate_stratum with sockbuf=0x115d0d0
[14:04:53] Stratum connect failed to pool 0: Failed to connect to 192.168.1.104 port 3333: Connection refused
[14:04:53] Initiate stratum failed
[14:04:53] Unable to get work from pool 0 stratum+tcp://192.168.1.104:3333
[14:04:53] No servers were found that could be used to get work from.
[14:04:53] Please check the details from the list below of the servers you have input
[14:04:53] Most likely you have input the wrong URL, forgotten to add a port, or have not set up workers
[14:04:53] Pool: 0  URL: stratum+tcp://192.168.1.104:3333  User: workeruser  Password: workerpass
[14:04:53] Press any key to exit or the miner will try again in 15s.

did lot of searches but without any result

**Comment by ghostlander** [comment]
If you're a pool op or just run a private pool, consider to enable GetWork and/or GetBlockTemplate in addition to Stratum if possible. For example, P2Pool works with GW just fine.
